GCC 8 warning in nightly builds

See [CDash](http://cdash.seqan.de/viewBuildError.php?type=1&buildid=289832)

```
/.../Darwin-16.7.0_g++-mp-8_32/include/seqan/basic/iterator_adaptor.h:102:9: warning: '<anonymous>' is used uninitialized in this function [-Wuninitialized]

         data_iterator = TIterator();
```

--- a/include/seqan/basic/iterator_adaptor.h
+++ b/include/seqan/basic/iterator_adaptor.h
@@ -96,11 +96,10 @@ public:
     // Constructors
     // ------------------------------------------------------------------------
 
-   
-    Iter() : data_container()
-    {
-        data_iterator = TIterator();
-    }
+
+    Iter() : data_container(),
+             data_iterator()
+    {}
 
     /*
     // TODO(doering): welches "begin" zur initialisierung von "data_iterator" aufrufen?
@@ -116,15 +115,13 @@ public:
     Iter(typename Parameter_<TContainer>::Type container_, TIterator it_)
             : data_container(_toPointer(container_)),
               data_iterator(it_)
-    {
-    }
+    {}
 
    
     Iter(Iter const & other_)
             : data_container(other_.data_container),
               data_iterator(other_.data_iterator)
-    {
-    }
+    {}
 
     // TODO(holtgrew): Use this technique to the other Iter specializations.
     template <typename TContainer_, typename TIterator2>

